collect.py

```diff
@@ -172,19 +167,22 @@ def do_something(driver, visited_paths, current_path, elem_properties=None):
             elems = get_elements_with_properties(driver, elem_properties, children)
             if len(elems) == 1:
                 elem = child
+                xpath = driver.execute_script(get_xpath_script, elem)
                 break
             else:
                 children_to_ignore.extend(elems)
-            visit(current_path, visited_paths)
-            break
     else:
-        if 'id' not in elem_properties['attributes'].keys():
+        if 'id' in elem_properties['attributes'].keys():
+            elem_id = elem_properties['attributes']['id']
+            elem = driver.find_element_by_id(elem_id)
+            xpath = driver.execute_script(get_xpath_script, elem)
+        elif xpath is not None:
+            elem = driver.find_element_by_xpath(xpath)
```

Comment:
If we can't find the element via xpath, we should fallback on finding it via the properties.

Reply:
Or maybe we should just assert that we found the element here.